vuu/src/test/scala/org/finos/vuu/viewport/VisualLinkedViewPortTest.scala

```diff
@@ -130,32 +134,48 @@ class VisualLinkedViewPortTest extends AbstractViewPortTestCase with Matchers wi
       viewPortContainer.runOnce()
 
-      Then("we should have nothing available in the viewport")
-      viewPortOrders.getKeys.length shouldEqual 0
+      Then("we should show all by default in the viewport")
+      viewPortOrders.getKeys.length shouldEqual 12
+
+      assertVpEqWithMeta(combineQs(viewPortOrders).filter(vpu => vpu.vp.id == viewPortOrders.id)) {
+        Table(
+          ("sel", "orderId", "trader", "ric", "tradeTime", "quantity"),
+          (0, "NYC-0000", "chris", "VOD.L", 1311544800000L, 100),
+          (0, "NYC-0001", "chris", "VOD.L", 1311544800010L, 101),
+          (0, "NYC-0002", "chris", "VOD.L", 1311544800020L, 102),
+          (0, "NYC-0003", "chris", "BT.L", 1311544800030L, 100),
+          (0, "NYC-0004", "chris", "BT.L", 1311544800040L, 101),
+          (0, "NYC-0005", "chris", "BT.L", 1311544800050L, 102),
+          (0, "NYC-0006", "chris", "BT.L", 1311544800060L, 103),
+          (0, "NYC-0007", "chris", "BP.L", 1311544800070L, 100),
+          (0, "NYC-0008", "chris", "BP.L", 1311544800080L, 101),
+          (0, "NYC-0009", "chris", "BP.L", 1311544800090L, 102)
+        )
+      }
 
       Then("Change the viewport to sort by quantity")
       viewPortContainer.change(RequestId.oneNew(), session, viewPortOrders.id, ViewPortRange(0, 10), vpcolumnsOrders, SortSpec(List(SortDef("quantity", 'D'))))
       viewPortContainer.changeSelection(session, outQueue, viewPortPrices.id, ViewPortSelectedIndices(Array(1)))
 
       viewPortContainer.runOnce()
 
-      Then("Check we still maintain the selection even with a new sort or filter")
-      assertVpEqWithMeta(combineQs(viewPortOrders).filter(vpu => vpu.vp.id == viewPortOrders.id)) {
-        Table(
-          ("sel"     ,"orderId" ,"trader"  ,"ric"     ,"tradeTime","quantity"),
-          (0         ,"NYC-0003","chris"   ,"BT.L"    ,1311544800030L,100       ),
-          (0         ,"NYC-0004","chris"   ,"BT.L"    ,1311544800040L,101       ),
-          (0         ,"NYC-0005","chris"   ,"BT.L"    ,1311544800050L,102       ),
-          (0         ,"NYC-0006","chris"   ,"BT.L"    ,1311544800060L,103       )
-        )
-      }
+      Then("check we now have 4 keys in the viewport")
+      viewPortOrders.getKeys.length shouldEqual 4
+
+//      Then("Check we still maintain the selection even with a new sort or filter")
+//      assertVpEqWithMeta(combineQs(viewPortOrders).filter(vpu => vpu.vp.id == viewPortOrders.id)) {
+//        Table(
+//          ("sel"     ,"orderId" ,"trader"  ,"ric"     ,"tradeTime","quantity"),
+//          (0         ,"NYC-0003","chris"   ,"BT.L"    ,1311544800030L,100       ),
+//          (0         ,"NYC-0004","chris"   ,"BT.L"    ,1311544800040L,101       ),
+//          (0         ,"NYC-0005","chris"   ,"BT.L"    ,1311544800050L,102       ),
+//          (0         ,"NYC-0006","chris"   ,"BT.L"    ,1311544800060L,103       )
+//        )
+//      }
```

Comment:
could we clean up commented code if we dont want it or uncomment and keep it? I think its trying to assert that changing the price table selection back to BT.L after sort, still apply the visual link filter on the orders table. Probably could be improved to make it clearer?